src/decode.h

```diff
@@ -1123,6 +1123,24 @@ static inline bool PacketIsNotTunnel(const Packet *p)
     return (p->ttype != PacketTunnelNone);
 }
 
+static inline bool VerdictTunnelPacketInternal(const Packet *p)
+{
+    const uint16_t outstanding = TUNNEL_PKT_TPR(p) - TUNNEL_PKT_RTV(p);
+    SCLogDebug("tunnel: outstanding %u", outstanding);
+
+    /* if there are packets outstanding, we won't verdict this one */
+    if (PacketIsTunnelRoot(p) && !PacketTunnelIsVerdicted(p) && !outstanding) {
+        SCLogDebug("root %p: verdict", p);
+        return true;
+    } else if (PacketIsTunnelChild(p) && outstanding == 1 && p->root &&
+               PacketTunnelIsVerdicted(p->root)) {
+        SCLogDebug("tunnel %p: verdict", p);
+    } else {
+        return false;
+    }
+    return true;
```

Comment:
nit: could it be
```
        return true;
    }
return false;
```
?

Reply:
Wouldn't Jason's suggestion and yours be conflicting, here?

Reply:
I think they'd both mean the same thing.

Reply:
Oh! I suck at this xD